sql/core/src/main/scala/org/apache/spark/sql/execution/datasources/rules.scala

```diff
@@ -78,9 +78,10 @@ class ResolveSQLOnFile(sparkSession: SparkSession) extends Rule[LogicalPlan] {
         case e: Exception =>
           // the provider is valid, but failed to create a logical plan
           u.failAnalysis(
-            errorClass = "_LEGACY_ERROR_TEMP_2332",
-            messageParameters = Map("msg" -> e.getMessage),
-            cause = e)
+            errorClass = "UNSUPPORTED_DATASOURCE",
```

Comment:
Is it correct error class? Don't you assigned the name `UNSUPPORTED_DATASOURCE_FOR_DIRECT_QUERY`?

Reply:
Oops... sorry I mistakenly didn't push the code changes.
Thanks for catching out!